src/hooks/useToast.tsx

```diff
@@ -0,0 +1,49 @@
+import React, { createContext, useContext, useState, useCallback } from 'react';
+import StoryToast from '../components/StoryToast';
+
+interface Toast {
+  id: number;
+  message: string;
+  type: 'success' | 'warning' | 'error' | 'info';
+}
+
+interface ToastContextType {
+  showToast: (message: string, type?: Toast['type'], duration?: number) => void;
+}
+
+const ToastContext = createContext<ToastContextType | undefined>(undefined);
+
+export const ToastProvider: React.FC<{ children: React.ReactNode }> = ({ children }) => {
+  const [toasts, setToasts] = useState<Toast[]>([]);
+
+  const showToast = useCallback((message: string, type: Toast['type'] = 'info', duration: number = 2000) => {
+    const id = Date.now();
+    const toast = { id, message, type };
+    setToasts((prevToasts) => [...prevToasts, toast]);
+
+    setTimeout(() => {
+      setToasts((prevToasts) => prevToasts.filter((t) => t.id !== id));
+    }, duration);
+  }, []);
+
+  return (
+    <ToastContext.Provider value={{ showToast }}>
+      {children}
+      <div style={{ position: 'fixed', top: 20, right: 20, zIndex: 1000 }}>
+        {toasts.map((toast) => (
+          <StoryToast key={toast.id} type={toast.type}>
+            {toast.message}
+          </StoryToast>
+        ))}
+      </div>
+    </ToastContext.Provider>
+  );
+};
+
+export const useToast = () => {
+  const context = useContext(ToastContext);
+  if (!context) {
+    throw new Error('useToast must be used within a ToastProvider');
```

Comment:
tutaj, w innym PR chcielibyśmy mieć jednak coś, co powoduje, że jeśli jednak nie ma otoczonego providerem (dla toastów), to 
1. sprawdzamy czy istnieje element o jakimś naszym hardcoded id
2. (jeśli nie, to dodajemy nowostworzony element do document.body, nadajemy mu id)
3. `toast-container` powinien mieć coś ala posision absolute lub fixed i powinien być pozycjonowany względem oczekiwanego umiejscowienia toastów
3. i `document.appendChild` każdego komunikatu dodanego za pomocą `useToast`